src/core/system_parameters/beidou_cnav2_navigation_message.cc

```diff
@@ -0,0 +1,646 @@
+/*!
+ * \file beidou_cnav2_navigation_message.cc
+ * \brief  Implementation of a beidou cnav2 Data message decoder as described in BEIDOU ICD
+ * \note Code added as part of GSoC 2018 program
+ * \author Dong Kyeong Lee, 2018. dole7890(at)colorado.edu
+ * \see <a href="http://m.beidou.gov.cn/xt/gfxz/201712/P020171226742357364174.pdf">BEIDOU ICD</a>
+ *
+ * -------------------------------------------------------------------------
+ *
+ * Copyright (C) 2010-2015  (see AUTHORS file for a list of contributors)
+ *
+ * GNSS-SDR is a software defined Global Navigation
+ *          Satellite Systems receiver
+ *
+ * This file is part of GNSS-SDR.
+ *
+ * GNSS-SDR is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * GNSS-SDR is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GNSS-SDR. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * -------------------------------------------------------------------------
+ */
+
+#include "beidou_cnav2_navigation_message.h"
+#include "gnss_satellite.h"
+#include <glog/logging.h>
+#include "../../algorithms/telemetry_decoder/libs/crc24q.h"
```

Comment:
Not sure if this is the right way of adding header files. The path should not be relative since cmakefiles.txt takes cares of this. Look for how the viterbi_decoder code is added and then reproduce the same calling format